frontend/app/routes/protected/person-case/personal-info.tsx

```diff
@@ -0,0 +1,290 @@
+        firstNamePreviouslyUsed: v.optional(v.array(v.string())),
+        lastNameAtBirth: v.pipe(
+          v.string(t('protected:personal-information.last-name-at-birth.required')),
+          v.trim(),
+          v.nonEmpty(t('protected:personal-information.last-name-at-birth.required')),
+          v.maxLength(
+            nameMaxLength,
+            t('protected:personal-information.last-name-at-birth.max-length', { maximum: nameMaxLength }),
+          ),
+          v.regex(REGEX_PATTERNS.NON_DIGIT, t('protected:personal-information.last-name-at-birth.format')),
+        ),
+        lastNamePreviouslyUsed: v.optional(v.array(v.string())),
+        gender: v.picklist(VALID_GENDER, t('protected:personal-information.gender.required')),
+      }) satisfies v.GenericSchema<PersonalInformationSessionData>;
+
+      const input = {
+        firstNamePreviouslyUsed: formData.getAll('firstNamePreviouslyUsed').map(String),
+        lastNameAtBirth: String(formData.get('lastNameAtBirth')),
+        lastNamePreviouslyUsed: formData.getAll('lastNamePreviouslyUsed').map(String),
+        gender: String(formData.get('gender')),
+      } satisfies Partial<PersonalInformationSessionData>;
+
+      const parseResult = v.safeParse(schema, input, { lang });
+
+      if (!parseResult.success) {
+        return data({ errors: v.flatten<typeof schema>(parseResult.issues).nested }, { status: 400 });
+      }
+
+      context.session.inPersonSINCase ??= {};
+      context.session.inPersonSINCase.personalInformation = parseResult.output;
+
+      throw i18nRedirect('routes/protected/person-case/birth-details.tsx', request);
+    }
+
+    default: {
+      throw new AppError(`Unrecognized action: ${action}`);
+    }
+  }
+}
+
+export default function PersonalInformation({ loaderData, params }: Route.ComponentProps) {
+  const { t } = useTranslation(handle.i18nNamespace);
+
+  const fetcherKey = useId();
+  const fetcher = useFetcher<Info['actionData']>({ key: fetcherKey });
+  const isSubmitting = fetcher.state !== 'idle';
+  const errors = fetcher.data?.errors;
+
+  const [firstName, setFirstName] = useState<string | undefined>(undefined);
+  const [firstNames, setFirstNames] = useState(loaderData.defaultFormValues?.firstNamePreviouslyUsed);
+  const [lastName, setLastName] = useState<string | undefined>(undefined);
+  const [lastNames, setLastNames] = useState(loaderData.defaultFormValues?.lastNamePreviouslyUsed);
+  const [srAnnouncement, setSrAnnouncement] = useState('');
+
+  const GENDER = ['female', 'male', 'other'] as const;
+
+  const genderOptions = GENDER.map((value) => ({
+    value: value,
+    children: t(`protected:personal-information.gender.options.${value}` as const),
+    defaultChecked: value === loaderData.defaultFormValues?.gender,
+  }));
+
+  const handleAddFirstName = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    if (firstName?.trim()) {
+      setFirstNames([...(firstNames ?? []), firstName.trim()]);
```

Comment:
When you're calling `setState()` with an existing value, you pass it in as a part of the function. So..

``` ts
setLastNames((prev) => [...(prev ?? []), firstname.trim()])
```

<https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state>